src/docfx/build/dependency/Resolve.cs

```diff
@@ -181,10 +183,21 @@ public static (Error error, string href, string display) TryResolveXref(string h
             var displayPropertyValue = xrefSpec.GetXrefPropertyValue(queries?["displayProperty"]);
             string display = !string.IsNullOrEmpty(displayPropertyValue) ? displayPropertyValue : (!string.IsNullOrEmpty(name) ? name : uid);
             var monikerQuery = !string.IsNullOrEmpty(moniker) ? $"view={moniker}" : "";
-            href = HrefUtility.MergeHref(xrefSpec.Href, monikerQuery, fragment.Length == 0 ? "" : fragment.Substring(1));
+
+            href = HrefUtility.MergeHref(referencedFile != null ? GetRelativeUrl(file, referencedFile) : xrefSpec.Href, monikerQuery, fragment.Length == 0 ? "" : fragment.Substring(1));
             return (null, href, display);
         }
 
+        /// <summary>
+        /// Get relative url from file to the file relative to
+        /// </summary>
+        public static string GetRelativeUrl(Document fileRelativeTo, Document file)
```

Comment:
nit: private

Reply:
it is used within `AttributeTransformer` as well since we rebase the link in the end